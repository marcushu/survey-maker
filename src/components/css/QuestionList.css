button.edDelBtn {
    color: rgb(74, 87, 104);
    background-color: transparent;
    border: none;
    margin-left: 15px;
}
