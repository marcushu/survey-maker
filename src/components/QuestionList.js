import React from 'react';
import PropTypes from 'prop-types';
import MultipleChoiceQuestion from './questions/MultipleChoiceQuestion';
import StringQuestion from './questions/StringQuestion';
import BinaryQuestion from './questions/BinaryQuestion';
import './css/QuestionList.css';

export default function QuestionList({ questions, deleteQuestion, editQuesiton, isEditable }) {

  const updateQuestion = updated => {
    const updateResult = old_question => {
      if (old_question.question === updated.question)
        old_question.result = updated.result;
    }

    questions.map(oldQuestion => updateResult(oldQuestion));
  }


  const editableOptions = question => {
    if (isEditable) return (
      <div>
        <button className="edDelBtn" onClick={() => deleteQuestion(question)}> Delete &nbsp;</button> | 
        <button className="edDelBtn" onClick={() => editQuesiton(question)}>Edit</button>
      </div>
    )
  }


  return (
    <div>
      {questions.map((question) =>
        <div key={question.question}>
          {question.questionType === "BINARY" &&
            <div>
              {editableOptions(question)}
              <BinaryQuestion
                question={question}
                updateQuestion={updateQuestion} />
              <br />
            </div>}
          {question.questionType === "MULTCHOICE" &&
            <div>
              {editableOptions(question)}
              <MultipleChoiceQuestion
                question={question}
                updateQuestion={updateQuestion} />
              <br />
            </div>}
          {question.questionType === "TEXT" &&
            <div>
              {editableOptions(question)}
              <StringQuestion
                question={question}
                updateQuestion={updateQuestion} />
              <br />
            </div>}
        </div>
      )}
    </div>
  )
}

QuestionList.propTypes = {
  questions: PropTypes.array,
  deleteQuestion: PropTypes.func,
  editQuesiton: PropTypes.func,
  isEditable: PropTypes.bool
}
