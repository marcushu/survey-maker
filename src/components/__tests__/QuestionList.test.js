import React from 'react';
import { render } from '@testing-library/react';
import QuestionList from '../QuestionList';

describe("QuestionList", () => {
    const mockFunc = jest.fn();

    const mockQuestions = [
        {
            choices: [],
            negativeIdentity: "No",
            positiveIdentity: "Yes",
            question: "Did you attend with a guest?",
            questionType: "BINARY",
            result: "",
            _id: "5fb43ce7f5397f164e5a7a0d"
        },
        {
            choices: [],
            negativeIdentity: "No",
            positiveIdentity: "Yes",
            question: "Did you stay the whole time?",
            questionType: "BINARY",
            result: "",
            _id: "5fb43ce7f5397f164e5a7a0e"
        },
        {
            choices: ["ok", "good", "great"],
            negativeIdentity: "",
            positiveIdentity: "",
            question: "How was the food?",
            questionType: "MULTCHOICE",
            result: "",
            _id: "5fb43ce7f5397f164e5a7a0f"
        },
        {
            choices: ["ok", "good", "great"],
            negativeIdentity: "",
            positiveIdentity: "",
            question: "How was the entertainment?",
            questionType: "MULTCHOICE",
            result: "",
            _id: "5fb43ce7f5397f164e5a7a10"
        },
        {
            choices: [],
            negativeIdentity: "",
            positiveIdentity: "",
            question: "How can we improve?",
            questionType: "TEXT",
            result: "",
            _id: "5fb43ce7f5397f164e5a7a11"
        }
    ];

    it("renders", () => {
        const { asFragment } = render(
            <QuestionList
                questions={mockQuestions}
                deleteQuestion={mockFunc}
                editQuestion={mockFunc}
                isEditable={true} />
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it("displays all the questions", () => {
        const { getByText } = render(
            <QuestionList
                questions={mockQuestions}
                deleteQuestion={mockFunc}
                editQuestion={mockFunc}
                isEditable={true} />
        );

        expect(getByText(/Did you attend with a guest?/i)).toBeInTheDocument();
        expect(getByText(/Did you stay the whole time?/i)).toBeInTheDocument();
        expect(getByText(/how was the food?/i)).toBeInTheDocument();
        expect(getByText(/How was the entertainment?/i)).toBeInTheDocument();
        expect(getByText(/How can we improve?/i)).toBeInTheDocument();
    })
});