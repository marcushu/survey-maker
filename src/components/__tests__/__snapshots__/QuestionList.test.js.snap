// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionList renders 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div>
        <div>
          <button
            class="edDelBtn"
          >
             Delete  
          </button>
           |
          <button
            class="edDelBtn"
          >
            Edit
          </button>
        </div>
        <div
          class="listItem d-flex justify-content-between"
        >
          <div
            class="question"
          >
            Did you attend with a guest?
          </div>
          <div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="Yes"
              />
              <label
                class="form-check-label"
                title=""
              >
                Yes
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="No"
              />
              <label
                class="form-check-label"
                title=""
              >
                No
              </label>
            </div>
          </div>
        </div>
        <br />
      </div>
    </div>
    <div>
      <div>
        <div>
          <button
            class="edDelBtn"
          >
             Delete  
          </button>
           |
          <button
            class="edDelBtn"
          >
            Edit
          </button>
        </div>
        <div
          class="listItem d-flex justify-content-between"
        >
          <div
            class="question"
          >
            Did you stay the whole time?
          </div>
          <div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="Yes"
              />
              <label
                class="form-check-label"
                title=""
              >
                Yes
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="No"
              />
              <label
                class="form-check-label"
                title=""
              >
                No
              </label>
            </div>
          </div>
        </div>
        <br />
      </div>
    </div>
    <div>
      <div>
        <div>
          <button
            class="edDelBtn"
          >
             Delete  
          </button>
           |
          <button
            class="edDelBtn"
          >
            Edit
          </button>
        </div>
        <div
          class="listItem d-flex justify-content-between"
        >
          <div
            class="question"
          >
            How was the food?
          </div>
          <div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="ok"
              />
              <label
                class="form-check-label"
                title=""
              >
                ok
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="good"
              />
              <label
                class="form-check-label"
                title=""
              >
                good
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="great"
              />
              <label
                class="form-check-label"
                title=""
              >
                great
              </label>
            </div>
          </div>
        </div>
        <br />
      </div>
    </div>
    <div>
      <div>
        <div>
          <button
            class="edDelBtn"
          >
             Delete  
          </button>
           |
          <button
            class="edDelBtn"
          >
            Edit
          </button>
        </div>
        <div
          class="listItem d-flex justify-content-between"
        >
          <div
            class="question"
          >
            How was the entertainment?
          </div>
          <div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="ok"
              />
              <label
                class="form-check-label"
                title=""
              >
                ok
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="good"
              />
              <label
                class="form-check-label"
                title=""
              >
                good
              </label>
            </div>
            <div
              class="form-check form-check-inline"
            >
              <input
                class="form-check-input"
                type="radio"
                value="great"
              />
              <label
                class="form-check-label"
                title=""
              >
                great
              </label>
            </div>
          </div>
        </div>
        <br />
      </div>
    </div>
    <div>
      <div>
        <div>
          <button
            class="edDelBtn"
          >
             Delete  
          </button>
           |
          <button
            class="edDelBtn"
          >
            Edit
          </button>
        </div>
        <div
          class="listItem"
        >
          <div
            class="question"
          >
            How can we improve?
          </div>
          <br />
          <textarea
            class="form-control"
            rows="3"
          />
        </div>
        <br />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
